dev/deps/spark-deps-hadoop-2.6

```diff
@@ -98,7 +98,7 @@ jackson-module-jaxb-annotations-2.6.7.jar
 jackson-module-paranamer-2.7.9.jar
 jackson-module-scala_2.11-2.6.7.1.jar
 jackson-xc-1.9.13.jar
-janino-3.0.8.jar
+janino-3.0.9.jar
```

Comment:
https://janino-compiler.github.io/janino/changelog.html

cc @kiszk @rednaxelafx Please check the changes made in 3.0.9

Reply:
It includes your fix https://github.com/janino-compiler/janino/pull/46

Reply:
Yep my primary reason is pulling in this fix for Scala 2.12: https://github.com/janino-compiler/janino/pull/54 The other changes I can see appear to be small bug fixes, some for Java 8 / 9 support, which is probably useful for us.

Reply:
Sorry for being late. I confirmed all of changes in the changelog. Two issues https://github.com/janino-compiler/janino/pull/54 and https://github.com/janino-compiler/janino/pull/46 are for Java 8 and later support. Others are bug fixes.

LGTM.